Develop3

ebatuev opened the pull request:
Создал отдельный репозиторий для тестов веб-приложения, с помощью Page Object.
Создал Maven-проект.
Подключил JUnit 4, Selenium, RestAssured и Allure.

Проект:
1. Cоздать класс PathAPI с эндпоинтами (ручками).

2. Cоздать классы для пользователя UserRegistration, UserLogin.
* Начать описывать - Сериализацию — преобразовать объект в JSON
*
* Как использовать сериализацию, если нужно отправить запрос с JSON:
*  1) Создать класс, поля которого соответствуют ключам JSON. Добавить конструктор с параметрами и конструктор без параметров.
*  2) В тесте создать объект класса с нужными значениями — через конструктор.
*  3) Сериализовать объект, то есть преобразовать его в JSON. Для этого нужно подключить библиотеку Gson.

3. Описать общие шаги для пользователя и вынести их в класс User.
   Класс являются наследниками класса PathAPI, где хранятся ручки.

4. Добавить класс с настройкой запуска браузера (Google Chrome и Яндекс.Браузер)

5. Создали page object — классы для страниц (RegistrationPage, LoginPage, MainPage, ProfilePage и ForgotPasswordPage) и описал элементы

6. Написал тестовые классы RegistrationUserTest, LoginUserTest и  NavigationTest.

7. Составил Allure-отчёт.

Commit:
1 коммит - тест

Commit:
2 коммит - тест

Commit:
3 коммит - добавил тестовые классы и отчет алюр

nikolay-977 commented on src/main/java/org/example/pages/ForgotPasswordPage.java:
```diff
@@ -0,0 +1,29 @@
+
+    // Методы:
+    // метод кликнуть на ссылку "Вспомнили пароль?"
+    public void clickOnEnterLinkOnForgotPasswordPage(){
```
Необходимо исправить: При реализации шагов используй аннотацию `@Step("Создание заказа")`. Без аннотации `@Step` шаг не отобразится в Allure отчете. В скобках указывай описание шага. Это сделает отчет Allure информативнее, что позволит сократить время локализации дефекта.

ebatuev replied:
Ценное замечание, спасибо)
Добавил аннотацию @Step в классы страниц с шагами (RegistrationPage, LoginPage, MainPage, ProfilePage и ForgotPasswordPage)

nikolay-977 commented on src/test/java/NavigationTest.java:
```diff
@@ -0,0 +1,54 @@
+        MainPage mainPage = new MainPage(getDriver());
+        mainPage.clickOnSaucesChapter();
+        mainPage.clickOnBunsChapter();
+        Assert.assertEquals("Неверный текст! Должны быть Булки","Булки", mainPage.getTextFromTheChosenElement()); // проверяет текст через метод assertEquals
```
Необходимо исправить: не следует выполнять все проверки в одном кейсе. Иначе в случае фейла первой проверки, последующие проверки не будут выполнены, это может аффектить другие тесты, что увеличит время локализации дефекта. Напиши отдельные кейсы для проверки каждого перехода.

ebatuev replied:
Написал отдельные кейсы для проверки каждого перехода

Commit:
4 коммит -
1) Добавил аннотацию @step в классы страниц с шагами (RegistrationPage, LoginPage, MainPage, ProfilePage и ForgotPasswordPage)
2) Написал отдельные кейсы для проверки каждого перехода

Commit:
5 коммит - Составил Allure-отчёт
